### PR TITLE
feat: add toast and react query

### PR DESCRIPTION
**Pull request by rizqitsani**
# Description & Technical Solution

- feat: add toast and react query

# Checklist

- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] Already rebased against main branch.

# Screenshots

[Toast-PPDB-Ultimate-Moon-Design-System.webm](https://user-images.githubusercontent.com/68275535/212282086-5b6b85bd-6893-4015-8dda-2bd13b9294c7.webm)

### CODE REVIEW

**Comment by ppdbultimate** [comment]
#### 📝 Changed routes:


| [**/_app.page**](https://design-system-moon-3e0iahx31-ppdbultimate.vercel.app/_app.page) | [**/index.page**](https://design-system-moon-3e0iahx31-ppdbultimate.vercel.app/index.page) |
|:-:|:-:|
|<a href="https://design-system-moon-3e0iahx31-ppdbultimate.vercel.app/_app.page">Screenshot of /_app.page</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/37jZopjr7NY1OspUdF6UETolkgArffdJFf-Hv9Lzb5L1DMbSjXGiJQcWFeff4yu7kCC7XxKldm7qnDRSbXHzAg.png">(view full size)</a> |<a href="https://design-system-moon-3e0iahx31-ppdbultimate.vercel.app/index.page">Screenshot of /index.page</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/O9YWoVKcTEZWw5_yKrNf6-AojSfacUdn5nLHJIkkGXuQUTWTpmNNbqiod4-f9PSz7N0Je45AfiI7qnev7HAfgw.png">(view full size)</a> |



| [**/sandbox/toast-rq.page**](https://design-system-moon-3e0iahx31-ppdbultimate.vercel.app/sandbox/toast-rq.page) |
|:-:|
|<a href="https://design-system-moon-3e0iahx31-ppdbultimate.vercel.app/sandbox/toast-rq.page">Screenshot of /sandbox/toast-rq.page</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/k8IqVgaUtceP0--lssdrPbZ8XpyhxVtp_6NPYj5DpDe1e-20vF0AZuoneEmGaaPFKkqbhZaWBV7wQGZkAr303A.png">(view full size)</a> |



And 2 other routes:
- [**/api/mock/login.api**](https://design-system-moon-3e0iahx31-ppdbultimate.vercel.app/api/mock/login.api)
- [**/api/mock/me.api**](https://design-system-moon-3e0iahx31-ppdbultimate.vercel.app/api/mock/me.api)



Commit 20370e8775dc17e5ec5a2f41879fa5c0cece826d (https://design-system-moon-3e0iahx31-ppdbultimate.vercel.app).

File: src/pages/sandbox/toast-rq.page.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  const { data: queryData } = useQueryToast(
+    useQuery<ApiReturn<User>, AxiosError<ApiError>>(['/me'], mockQuery)
+  );
+
+  return (
+    <>
+      <Seo templateTitle='Toast' />
+
+      <section className='bg-gray-100'>
```
Comment (theodorusclarence) [inline src/pages/sandbox/toast-rq.page.tsx:44]: ini white ajaa bgnyaa

File: src/pages/sandbox/toast-rq.page.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  } = useMutationToast<ApiReturn<undefined>, LoginData>(
+    useMutation((data) => apiMock.post('/login', data).then((res) => res.data))
+  );
+
+  const { data: queryData } = useQueryToast(
+    useQuery<ApiReturn<User>, AxiosError<ApiError>>(['/me'], mockQuery)
+  );
+
+  return (
+    <>
```
Comment (theodorusclarence) [inline src/pages/sandbox/toast-rq.page.tsx:41]: sama ini kayaknya Layoutnya belom masuk wkwkwkwk

File: src/pages/sandbox/toast-rq.page.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      <Seo templateTitle='Toast' />
+
+      <section className='bg-gray-100'>
+        <div className='layout flex min-h-screen flex-col items-start space-y-3 py-20'>
+          <Typography variant='h2' as='h2'>
```
Comment (theodorusclarence) [inline src/pages/sandbox/toast-rq.page.tsx:46]: ini kalo bisa h1 biar konsisten sama yang lain

**Review by theodorusclarence — CHANGES_REQUESTED** [review]

**Comment by winatungmiharja** [comment]
Mas ini hover nya boleh bg nya diganti light aja biar gak terlalu gelap
image

**Review by winatungmiharja — APPROVED** [review]

**Review by theodorusclarence — APPROVED** [review]

**Pull request merged**